docs/11_search.md

```diff
@@ -0,0 +1,286 @@
+- ExecutionTime is for Retry/Cron user to query workflow that will run in future
+
+To list only open workflows, add `CloseTime = missing` to the end of query.
+
+If you use retry or cron feature, to query workflows that will start execution in certain time range, you can add predicates on ExecutionTime. For example. `ExecutionTime > 2019-01-01T10:00:00-07:00`. Note that if predicates on ExecutionTime is included, only cron or workflow that needs retry will be returned.
+
+### General notes about queries
+
+- Pagesize is default to 1000, and can not be larger than 10k
+- Range query on Cadence timestamp (StartTime, CloseTime, ExecutionTime) cannot be larger than 9223372036854775807 (maxInt64 - 1001)
+- Query by time range will have 1ms resolution
+- Query column names are case sensitive
+- ListWorkflow may take longer time when retrieve large number of workflow (10M+)
+- To retrieve a large number of workflows without caring about order, use the ScanWorkflow API
+- To efficiently count the number of workflows, use the CountWorkflow API
+
+## Tools support
+
+### CLI
+
+Support for search attributes is available as of version 0.6.0 of the Cadence server. You can also use the CLI from the latest [CLI Docker image](https://hub.docker.com/r/ubercadence/cli) (supported on 0.6.4 or later).
+
+#### Start workflow with search attributes
+
+```bash
+cadence --do samples-domain workflow start --tl helloWorldGroup --wt main.Workflow --et 60 --dt 10 -i '"vancexu"' -search_attr_key 'CustomIntField | CustomKeywordField | CustomStringField |  CustomBoolField | CustomDatetimeField' -search_attr_value '5 | keyword1 | vancexu test | true | 2019-06-07T16:16:36-08:00'
+```
+
+#### Search workflows with new list API
+
+```bash
+cadence --do samples-domain wf list -q '(CustomKeywordField = "keyword1" and CustomIntField >= 5) or CustomKeywordField = "keyword2"' -psa
+```
+
+```bash
+cadence --do samples-domain wf list -q 'CustomKeywordField in ("keyword2", "keyword1") and CustomIntField >= 5 and CloseTime between "2018-06-07T16:16:36-08:00" and "2019-06-07T16:46:34-08:00" order by CustomDatetimeField desc' -psa
+```
+
+To list only open workflows, add `CloseTime = missing` to the end of the query.
+
+Note that queries can support more than one type of filter:
+
+```bash
+cadence --do samples-domain wf list -q 'WorkflowType = "main.Workflow" and (WorkflowID = "1645a588-4772-4dab-b276-5f9db108b3a8" or RunID = "be66519b-5f09-40cd-b2e8-20e4106244dc")'
+```
+
+```bash
+cadence --do samples-domain wf list -q 'WorkflowType = "main.Workflow" StartTime > "2019-06-07T16:46:34-08:00" and CloseTime = missing'
+```
+
+### Web UI support
+
+Queries are supported in [Cadence Web](https://github.com/uber/cadence-web) as of release 3.4.0. Use the "Basic/Advanced" button to switch to "Advanced" mode and enter the query in the search box.
+
+## Local Testing
+
+1. Increase Docker memory to higher 6GB. Docker -> Preference -> advanced -> memory limit
```

Comment:
higher 6GB -> higher than 6GB (?)

Docker -> Preference -> advanced -> memory limit =>
Navigate to Docker -> Preferences -> Advanced -> Memory
(this according to my UI; just upgraded Docker)